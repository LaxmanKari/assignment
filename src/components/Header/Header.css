.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 20px;
  background-color: #ffffff;
}

.header-left {
  display: flex;
  align-items: center;
}

.menu-button {
  background: none;
  border: none;
  font-size: 20px;
  cursor: pointer;
  margin-right: 20px;
  padding: 5px;
  color: #333333;
}

.menu-button.active {
  background-color: #000000;
  color: #ffffff;
  border-radius: 4px;
}

.menu-button.active .fa-bars::before {
  color: #ffffff;
}

.menu-button:not(.active) .fa-bars::before {
  color: #333333;
}

.menu-item {
  background: none;
  border: none;
  font-size: 14px;
  cursor: pointer;
  padding: 5px 10px;
  margin-left: 10px;
  color: #333333;
}

.menu-item:hover {
  background-color: #f0f0f0;
}

.header-right {
  display: flex;
  align-items: center;
  gap: 20px;
}

.icon-button {
  background: none;
  border: none;
  color: #333333;
  font-size: 20px;
  cursor: pointer;
}

.login-button {
    background-color: white;
    color: black; 
    border: none;
    padding: 5px 10px;
    border-radius: 5px;
    cursor: pointer;
}
