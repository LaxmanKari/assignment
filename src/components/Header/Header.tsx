import React, { useState } from "react";
import './header.css';
import { useNavigate } from "react-router-dom";

function Header() {
  const [isMenuOpen, setMenuOpen] = useState<boolean>(true);
  const navigate = useNavigate();
  const toggleMenu = () => {
    setMenuOpen(!isMenuOpen);
  };

  const handleHomeClick = () => {
    navigate("/");
  }
  return (
    <header className="header">
      <div className="header-left">
        <button
          className={`menu-button ${isMenuOpen ? "active" : ""}`}
          onClick={toggleMenu}
          data-testid="menu-button"
        >
          <i className="fas fa-bars"></i>
        </button>
        {isMenuOpen && (
          <>
            <button className="menu-item" onClick={handleHomeClick}>HOME</button>
            <button className="menu-item">SHOP</button>
            <button className="menu-item">MAGAZINE</button>
          </>
        )}
      </div>
      <div className="header-right">
        <button className="icon-button " data-testid="search-button">
          <i className="fas fa-search"></i>
        </button>
        <button className="icon-button" data-testid="cart-button">
          <i className="fas fa-shopping-cart"></i>
        </button>
        <button className="login-button">LOGIN</button>
      </div>
    </header>
  );
}

export default Header;
