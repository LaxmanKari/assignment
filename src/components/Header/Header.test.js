import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Header from "./Header";
// qqimport "@testing-library/jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => jest.fn(),
}));

const renderWithRouter = () => {
  return render(
    <Router>
      <Header />
    </Router>
  );
};

describe("Header Component", () => {
  test("renders Header component correctly and matches snapshot", () => {
    const { asFragment } = renderWithRouter();
    expect(asFragment()).toMatchSnapshot();
  });

  test("should toggle menu visibility when menu button is clicked", () => {
    renderWithRouter();

    const menuButton = screen.getByTestId("menu-button");

    expect(screen.getByText("HOME")).toBeVisible();
    expect(screen.getByText("SHOP")).toBeVisible();
    expect(screen.getByText("MAGAZINE")).toBeVisible();

    fireEvent.click(menuButton);

    expect(screen.queryByText("HOME")).toBe(null);
    expect(screen.queryByText("SHOP")).toBe(null);
    expect(screen.queryByText("MAGAZINE")).toBe(null);

    fireEvent.click(menuButton);

    expect(screen.getByText("HOME")).toBeVisible();
    expect(screen.getByText("SHOP")).toBeVisible();
    expect(screen.getByText("MAGAZINE")).toBeVisible();
  });

  test("should navigate to home page when HOME button is clicked", () => {
    const navigate = jest.fn();

    jest
      .spyOn(require("react-router-dom"), "useNavigate")
      .mockReturnValue(navigate);

    renderWithRouter();

    fireEvent.click(screen.getByText("HOME"));

    expect(navigate).toHaveBeenCalledWith("/");
  });

  test("should render login button and icon buttons correctly", () => {
    renderWithRouter();
    expect(screen.getByText("LOGIN")).toBeVisible();
    expect(screen.getByTestId("search-button")).toBeVisible();
    expect(screen.getByTestId("cart-button")).toBeVisible();
  });
});
