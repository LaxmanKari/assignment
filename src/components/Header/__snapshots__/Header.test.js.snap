// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders Header component correctly and matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="header-left"
    >
      <button
        class="menu-button active"
        data-testid="menu-button"
      >
        <i
          class="fas fa-bars"
        />
      </button>
      <button
        class="menu-item"
      >
        HOME
      </button>
      <button
        class="menu-item"
      >
        SHOP
      </button>
      <button
        class="menu-item"
      >
        MAGAZINE
      </button>
    </div>
    <div
      class="header-right"
    >
      <button
        class="icon-button "
        data-testid="search-button"
      >
        <i
          class="fas fa-search"
        />
      </button>
      <button
        class="icon-button"
        data-testid="cart-button"
      >
        <i
          class="fas fa-shopping-cart"
        />
      </button>
      <button
        class="login-button"
      >
        LOGIN
      </button>
    </div>
  </header>
</DocumentFragment>
`;
